pkg/service/orchestrator/rerecord.go

```diff
@@ -17,7 +17,34 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+func (o *Orchestrator) checkForTemplates(ctx context.Context) {
+	// Check if the testcases are already templatized.
+	var notTemplatized []string
+	for testSet := range o.config.Test.SelectedTests {
+		conf, err := o.replay.GetTestSetConf(ctx, testSet)
+		if err != nil || conf == nil || conf.Template == nil {
+			notTemplatized = append(notTemplatized, testSet)
+		}
+	}
+	if len(notTemplatized) > 0 {
+		o.logger.Warn("The following testSets are not templatized. Do you want to templatize them to handle noisy fields?(y/n)", zap.Any("testSets:", notTemplatized))
+		reader := bufio.NewReader(os.Stdin)
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			o.logger.Warn("Failed to read input. Skipping templatization")
+		}
+		if input == "n\n" || input == "N\n" {
+			o.logger.Info("Skipping templatization")
+		} else if input == "y\n" || input == "Y\n" {
+			if err := o.replay.Templatize(ctx, notTemplatized); err != nil {
```

Comment:
Use the config rather than passing it as an argument. `r.config.Templatize.TestSets`. And Modify the function definition of templatize from `Templatize(ctx context.Context, testSets []string)` to just `Templatize(ctx context.Context)`

Reply:
@PranshuSrivastava Please address this comment as well.